lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -819,6 +829,19 @@ def test_view_certificate_link_hidden(self):
         resp = views.progress(self.request, course_id=unicode(self.course.id))
         self.assertContains(resp, u"Download Your Certificate")
 
+    def test_query_count_instructor_led(self):
+        """Test that query counts remain the same for self-paced and instructor-led courses."""
+        with self.assertNumQueries(18), check_mongo_calls(4):
+            resp = views.progress(self.request, course_id=unicode(self.course.id))
+        self.assertEqual(resp.status_code, 200)
+
+    def test_query_count_self_paced(self):
+        """Test that query counts remain the same for self-paced and instructor-led courses."""
+        self.setup_course(self_paced=True)
+        with self.assertNumQueries(18), check_mongo_calls(4):
+            resp = views.progress(self.request, course_id=unicode(self.course.id))
+        self.assertEqual(resp.status_code, 200)
```

Comment:
The comparison between these two would be clearer if you used `ddt`, so that you could pass the `self_paced` status, and then the query counts, and then run a single test.